resources/guidelines/testing/store/quality-guidelines-plugins/README.md

```diff
@@ -0,0 +1,644 @@
+
+Apart from that, you can validate the required credentials while saving them in the app settings. In this case, a status message must be displayed in the backend and Shopware log.
+
+[Example for implementing an API Test Button into the System Config form](https://github.com/shyim/ShyimApiTest)
+
+### Automatic code reviews with PhpStan and SonarQube
+
+Our most current code review configurations that we use when uploading apps via the Shopware Account can be found on GitHub.
+
+Refer to [code reviews for Shopware 5 on GitHub](https://github.com/shopwareLabs/store-plugin-codereview)
+
+### Automated code tests with Cypress
+
+There are Cypress tests for Shopware 5 and 6 on GitHub. The project is driven by the Friends of Shopware group. You can contribute at any time:
+
+*   [Cypress Tests for Shopware 5](https://github.com/FriendsOfShopware/FroshCypress)
+
+### Helpful tools for app developers
+
+* [FroshPluginUploader](https://github.com/FriendsOfShopware/FroshPluginUploader): Tool for validating and uploading new SW6 app releases to the Community Store (GitHub Project from "Friends of Shopware")]
+* [Shopware CLI tools](https://github.com/shopwareLabs/sw-cli-tools): When you think about performance, these are various useful console helpers for generating data.
+
+### Shopware 5.0 SDK License
+
+The Shopware 5 backend is completely based on the Ext JS framework by Sencha ([www.sencha.com](http://www.sencha.com/)). Suppose you are developing Shopware 5 apps that are built upon or alter the Shopware/ExtJS layer and wish to offer them with a proprietary license (apps that are not licensed under the GNU Affero General Public License version 3 or a compatible license); you require a Shopware SDK license.
+
+Thanks to an OEM agreement with Sencha, Shopware is able to offer this license on a per developer basis, allowing the license holder to use this framework for their own apps.
+
+**When will I not need a SW5 SDK-License?**
+
+1. When app will be sold and uses no extJS
+
+2. When app is for free and uses extJS
+
+3. When app is for free and uses no extJS
+
+If you need a license, please write an email to [alliances@shopware.com](mailto:alliances@shopware.com?subject=extJS-SDK%20needed).
+
+## Automatic code review - Errors
+
+### The required composer.json file was not found."
+
+**Cause:** Error in composer.json  
+
+One possible cause is that the technical app name from the Community Store or account does not match the technical name entered in composer.json, or the app is incorrectly zipped.  
+The technical app name had to be stored in the last part of the composer.json: composore.json > extra > shopware-plugin-class. So take a look at the bootstrap class. Most of the error were caused by the wrong technical name. For example, "Swag\\MyPlugin\\SwagMyPluginSW6" instead of "Swag\\MyPlugin\\SwagMyPlugin".
```

Comment:
composer.json: composore.json > extra > shopware-plugin-class -> is this a syntax ?

Reply:
no this is to mirror the folder structure